src/message.c

```diff
@@ -787,44 +787,103 @@ static void send_handler(void *context, const unsigned char *start, int length)
     pn_link_send(pnl, (const char*) start, length);
 }
 
+/**
+ * Returns true if the data backing the iterator matches any of the router specific annotations like "x-opt-qd.ingress" or
+ * "x-opt-qd.trace" or "x-opt-qd.to" or "x-opt-qd.phase
+ */
+static bool qd_message_is_router_annotation(qd_field_iterator_t *iter)
```

Comment:
I failed to read the doc for qd_field_iterator_prefix. I have used it now instead of qd_message_is_router_annotation